src/test/java/org/jenkinsci/plugins/durabletask/BourneShellScriptTest.java

```diff
@@ -285,53 +394,44 @@ public void smokeTest() throws Exception {
         c.cleanup(ws);
     }
 
-    private void setGlobalInterpreter(String interpreter) {
-        j.jenkins.getInjector().getInstance(Shell.DescriptorImpl.class).setShell(interpreter);
+    /**
+     * Checks if the golang binary outputs to stdout under normal shell execution.
+     * The binary must NOT output to stdout or else it will crash when Jenkins is terminated
+     * unexpectedly.
+     */
+    @Test public void stdout() throws Exception {
+        ByteArrayOutputStream baos = new ByteArrayOutputStream();
+        TeeOutputStream teeOut = new TeeOutputStream(baos, System.out);
+        StreamTaskListener stdoutListener = new StreamTaskListener(teeOut, Charset.defaultCharset());
+        String script = String.format("echo hello world");
+        Controller c = new BourneShellScript(script).launch(new EnvVars(), ws, launcher, stdoutListener);
+        awaitCompletion(c);
+        assertThat(baos.toString(), isEmptyString());
+        c.cleanup(ws);
     }
 
-    private void awaitCompletion(Controller c) throws IOException, InterruptedException {
-        while (c.exitStatus(ws, launcher, listener) == null) {
-            Thread.sleep(100);
+    @Issue("JENKINS-58290")
+    @Test public void backgroundLaunch() throws IOException, InterruptedException {
```

Comment:
The only test that appears to be failing is the `NATIVE` version for `backgroundLaunch`. This is just running the test locally on the agent.
https://ci.jenkins.io/job/Plugins/job/durable-task-plugin/job/PR-103/4/testReport/junit/org.jenkinsci.plugins.durabletask/BourneShellScriptTest/linux_8___Archive__linux_8____backgroundLaunch_2__NATIVE_/
I'm not sure if this is because of the agent or not, but since all the other tests pass locally for the agent, I'm inclined to just disable `backgroundLaunch` for the `NATIVE` platform